File: geocode/geocode.js
const request = require('request');

const geocodeAddress = (address, callback) => {

  const encodedInputAddress = encodeURIComponent(address);

  request({
    url: `https://maps.googleapis.com/maps/api/geocode/json?address=${encodedInputAddress}`,
    json: true
  }, (err, res, body) => {
    if (err) {
      callback('\nUnable to connect to Google servers!\n')
    } else if (body.status === 'ZERO_RESULTS') {
      callback('\nUnable to find specified address!\n');
    } else if (body.status === 'OK') {
      callback(undefined, {
        address: body.results[0].formatted_address,
        latitude: body.results[0].geometry.location.lat,
        longitude: body.results[0].geometry.location.lng
      });
    }
  });

};

module.exports.geocodeAddress = geocodeAddress;

File: python/geocode.py
# ./weather-app/python/geocode.py
import requests

from urllib.parse import quote

def getLatLon(inputAddress):
    '''
    Function to fetch the latitude, longitude from the Google Maps API
    using tne stdin of 'inputAddress'
    '''
    url = 'https://maps.googleapis.com/maps/api/geocode/json?address='
    encodedInputAddress = quote(inputAddress)
    apiAddress = url + encodedInputAddress

    r = requests.get(apiAddress)
    lat = r.json()['results'][0]['geometry']['location']['lat']
    lon = r.json()['results'][0]['geometry']['location']['lng']
    return lat, lon
